server_apis/python/example/auto_blacklist_from_watchlist.py

```diff
@@ -0,0 +1,162 @@
+                
+                parsed_json = json.loads(body)
+                lst = parsed_json['docs']
+                for item in lst:
+                    blacklist_binary(item['process_md5'])
+
+    except Exception, e:
+        print e
+    finally:
+        # need to make sure we ack the messages so they don't get left un-acked in the queue
+        # we set multiple to true to ensure that we ack all previous messages
+        channel.basic_ack(delivery_tag=method_frame.delivery_tag, multiple=True)
+
+    return
+
+def generate_queue_name():
+    """
+    generates a random queue name
+    """
+    return str(random.randint(0,10000)) + "-" + str(random.randint(0,100000))
+
+def build_cli_parser():
+
+    parser = optparse.OptionParser(usage="%prog [options]", 
+        description="Example CBSAPI script to consume published events")
+    
+    parser.add_option("-p", "--password", action="store", default="", dest="password",
+        help="RabbitMQ password; see /etc/cb/cb.conf\nDefault is an empty string")
+    
+    parser.add_option("-u", "--usename", action="store", default="cb", dest="username",
+        help="RabbitMQ username; see /etc/cb/cb.conf\nDefault is cb")
+    
+    parser.add_option("-s", "--cbserverip", action="store", default="localhost", dest="cbserver", 
+        help="Carbon Black server IP for rabbitmq server\nDefault is localhost")
+    
+    parser.add_option("-t", "--cbtoken", action="store", default="", dest="cbtoken",
+        help='Carbon Black API token\nDefault is an empty string')
+
+    return parser
+
+if __name__ == "__main__":
+
+    #    
+    # build the command line parser and ensure the required password option was provided
+    #
+    parser = build_cli_parser()
+    (options, args) = parser.parse_args()
+
+    global cbtoken 
+    cbtoken = options.cbtoken
+    global cbserver
+    cbserver = options.cbserver
+    
+    #
+    # Set the connection parameters to connect to to the rabbitmq:5004
+    # using the supplied username and password
+    #
+    credentials = pika.PlainCredentials(options.username, 
+        options.password)
+    
+    #
+    # Create our parameters for pika
+    #
+    parameters = pika.ConnectionParameters(options.cbserver,
+                                           5004,
+                                           '/',
+                                           credentials)
+
+    #
+    # Create the connection
+    #
+    connection = pika.BlockingConnection(parameters)
+
+    #
+    # Get the channel from the connection
+    #
+    channel = connection.channel()
+
+    #
+    # Create a random queue name
+    #
+    queue_name = generate_queue_name()
+
+    #
+    # make sure you use auto_delete so the queue isn't left filling
+    # with events when this program exists.
+    channel.queue_declare(queue=queue_name, auto_delete=True)
+
+    channel.queue_bind(exchange='api.events', queue=queue_name, routing_key='#')
```

Comment:
we can change the routing_key to filter on only watchlist events: watchlist.hit.# will match anything that starts with watchlist.